src/Common/BufferAllocationPolicy.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "config.h"
+
+#include "logger_useful.h"
+
+#include <list>
+
+namespace DB
+{
+
+struct BufferAllocationSettings
+{
+    size_t strict_upload_part_size = 0;
```

Comment:
Since now this struct lives in `src/Common` and related to any buffer allocation at all, I suggest renaming these fields by removing `upload` or `part`, or `single` substrings.

Reply:
ya true, I will remove upload & part from some places. I feel we still need it for size to specify its only a part size not the complete size. Here are the proposed new names, let me know your thoughts

```
strict_upload_part_size. -> strict_part_size
min_upload_part_size  -> min_part_size
max_upload_part_size -> max_part_size
upload_part_size_multiply_factor -> multiply_factor
upload_part_size_multiply_parts_count_threshold -> multiply_parts_count_threshold
max_single_part_upload_size -> max_single_part_size
```

Reply:
addressed it